token/services/network/common/rws/translator/rwset.go

```diff
@@ -6,11 +6,184 @@ SPDX-License-Identifier: Apache-2.0
 
 package translator
 
+import (
+	"crypto/sha256"
+
+	"github.com/gobuffalo/packr/v2/file/resolver/encoding/hex"
+	"github.com/hyperledger-labs/fabric-smart-client/pkg/utils/errors"
+)
+
 //go:generate counterfeiter -o mock/rwset.go -fake-name RWSet . RWSet
 
+type KeyTranslator interface {
+	CreateTokenRequestKey(id string) (string, error)
+	CreateSetupKey() (string, error)
+	CreateSetupHashKey() (string, error)
+	CreateOutputKey(id string, index uint64) (string, error)
+	CreateOutputSNKey(id string, index uint64, output []byte) (string, error)
+	CreateInputSNKey(id string) (string, error)
+	CreateIssueActionMetadataKey(key string) (string, error)
+	// CreateTransferActionMetadataKey returns the transfer action metadata key built from the passed
+	// transaction id, subkey, and index. Index is used to make sure the key is unique with the respect to the
+	// token request this key appears.
+	CreateTransferActionMetadataKey(key string) (string, error)
+	GetTransferMetadataSubKey(k string) (string, error)
+}
+
 // RWSet interface, used to read from, and write to, a rwset.
 type RWSet interface {
 	SetState(namespace string, key string, value []byte) error
 	GetState(namespace string, key string) ([]byte, error)
 	DeleteState(namespace string, key string) error
 }
+
+// KeyVersion defines the supported key versions
+type KeyVersion = int
```

Comment:
Maybe we can call it SupportedKeyVersion, to distinguish it from the actual KeyVersion.

Reply:
okay, but I'll change the comment